### ISSUE TITLE
Use Sparkle for updating under Windows and OSX

### ISSUE THREAD

**Issue by x-a-n-a-x** [OpenRA/OpenRA#9206]
Here is an application update framework for Windows, WinSparkle :
http://winsparkle.org
https://github.com/vslavik/winsparkle

... which is inspired by Sparkle for OS X : http://sparkle-project.org/

**Comment by pchote** [comment]
There are three parts to this:
- [ ] Modifying our packaging scripts and [OpenRAWeb](https://github.com/OpenRA/OpenRAWeb) to generate and publish an appcast (fiddly).
- [ ] Adding the Sparkle framework to the OSX launcher application (easy).
- [ ] Adding the WinSparkle framework to the Windows launcher application (not sure, but probably easy).

**Comment by LavenderMoon** [comment]
Will Linux be left out in this?

**Comment by pchote** [comment]
Fedora/OpenSUSE/Arch/Gentoo/FreeBSD already auto-update via their respective repositories.
Ubuntu is covered by #9201.

**Comment by LavenderMoon** [comment]
Ah, I didn't consider that.

**Comment by Herve-M** [comment]
I am on it, I am just waiting that my .Net Wrapper PR is accepted into WinSparkle. (https://github.com/vslavik/winsparkle/pull/73). After that, I will propose a Nuget package to ease deployment of dependencies.

**Comment by Mailaender** [comment]
There is already https://npackd.appspot.com/p/openra by @tim-lebedkov.

**Comment by Phrohdoh** [comment]
Npackd looks to be a package manager (similar to choco) while Sparkle and Co. are (auto-)updaters.

**Comment by x-a-n-a-x** [comment]
@Herve-M : got news for us? Have you got time to work on its integration in OpenRA ?

**Comment by Herve-M** [comment]
@x-a-n-a-x :
A little todo list about my advancement: 

- SparkleNet :
- [ ] Redo example
- [ ] Correct miss-coding
- [ ] Add nugget package  
- OpenRA : 
I have done most of integration of the SparkleNet into project, only dependency deployment is missing.  
- OpenRA.Web :
- [ ] Generation of appcast

**Comment by PunkPun** [comment]
Is this still relevant?

**Comment by pchote** [comment]
Yes, it would still be nice to have this option on at least macOS.

**Comment by dragunoff** [comment]
AppImages also have update mechanisms similar to Sparkle: https://docs.appimage.org/packaging-guide/optional/updates.html#making-appimages-self-updateable
